### ISSUE TITLE
Same "root" id for different fieldset or input 

### ISSUE THREAD

**Issue by jnachtigall** [rjsf-team/react-jsonschema-form#1824]
### Description

The default bootstrap/core theme renders multiple `id` with the default root. This seems to be caused by https://github.com/rjsf-team/react-jsonschema-form/blob/75f734d8ec7beb29e60ac27ceaf6deadd2a71734/packages/core/src/components/fields/ObjectField.js#L33

Unfortunately, using a `ObjectFieldTemplate` it is not easy to override the `id` using the `idPrefix` and property `name` because both are not passed to the `ObjectFieldTemplate`. So no easy workaround afais, instead needs a fix here. FWIW, btw I think it would be great if `ObjectField` would also pass its `name` to `ObjectFieldTemplate`.

### Steps to Reproduce

1. Go to this [playground](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJhbmdhYmVuWnVyTmllZGVybGFzc3VuZ0Fic2Nobml0dCI6eyJhcnREZXJOaWVkZXJsYXNzdW5nQWJzY2huaXR0Ijp7ImFuZ2FiZW5adXJIYXVwdG5pZWRlcmxhc3N1bmdBYnNjaG5pdHQiOnt9fX0sImFuZ2FiZW5adXJBcnREZXNCZXRyaWViZXNBYnNjaG5pdHQiOnt9LCJhbmdhYmVuWnVyUXVhbGlmaWthdGlvbkFic2Nobml0dCI6eyJoYW5kd2Vya3NrYXJ0ZUFic2Nobml0dCI6eyJoYW5kd2Vya3NrYXJ0ZVZvcmhhbmRlbkFic2Nobml0dCI6e30sImhhbmR3ZXJrc2thcnRlTmljaHRWb3JoYW5kZW5BYnNjaG5pdHQiOnt9fX0sImVybGF1Ym5pc0Fic2Nobml0dCI6eyJlcmxhdWJuaXNWb3JoYW5kZW5BYnNjaG5pdHQiOnt9LCJlcmxhdWJuaXNOaWNodFZvcmhhbmRlbkFic2Nobml0dCI6e319LCJmaXJzdE5hbWUiOiJDaHVjayIsImxhc3ROYW1lIjoiTm9ycmlzIiwiYWdlIjo3NSwiYmlvIjoiUm91bmRob3VzZSBraWNraW5nIGFzc2VzIHNpbmNlIDE5NDAiLCJwYXNzd29yZCI6Im5vbmVlZCJ9LCJzY2hlbWEiOnsiJHNjaGVtYSI6Imh0dHA6Ly9qc29uLXNjaGVtYS5vcmcvZHJhZnQtMDcvc2NoZW1hIyIsInByb3BlcnRpZXMiOnsiYW5nYWJlblp1ck5pZWRlcmxhc3N1bmdBYnNjaG5pdHQiOnsidGl0bGUiOiJBbmdhYmVuIHp1ciBOaWVkZXJsYXNzdW5nIiwidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnt9fSwiYW5nYWJlblp1ckFydERlc0JldHJpZWJlc0Fic2Nobml0dCI6eyJ0aXRsZSI6IkFydCBkZXMgQmV0cmllYmVzIiwidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnt9LCJyZXF1aXJlZCI6WyJhcnREZXNCZXRyaWViZXMiLCJpc3ROZWJlbmVyd2VyYiIsImFuemFobFZvbGx6ZWl0QmVzY2hhZWZ0aWd0ZSIsImFuemFobFRlaWx6ZWl0YmVzY2jkZnRpZ3RlIl19LCJhbmdhYmVuWnVyUXVhbGlmaWthdGlvbkFic2Nobml0dCI6eyJ0aXRsZSI6IkFuZ2FiZW4genUgSWhyZXIgUXVhbGlmaWthdGlvbiIsInR5cGUiOiJvYmplY3QiLCJwcm9wZXJ0aWVzIjp7fX0sImVybGF1Ym5pc0Fic2Nobml0dCI6eyJ0aXRsZSI6IkVybGF1Ym5pcyIsInR5cGUiOiJvYmplY3QiLCJwcm9wZXJ0aWVzIjp7fSwicmVxdWlyZWQiOlsiTGllZ3REaWVFcmxhdWJuaXNWb3IiXX0sInBmbGljaHRmZWxkZXJIaW53ZWlzIjp7InRpdGxlIjoiVGVzdCBzdHJpbmciLCJ0eXBlIjoic3RyaW5nIn19fSwidWlTY2hlbWEiOnsiZmlyc3ROYW1lIjp7InVpOmF1dG9mb2N1cyI6dHJ1ZSwidWk6ZW1wdHlWYWx1ZSI6IiIsInVpOmF1dG9jb21wbGV0ZSI6ImZhbWlseS1uYW1lIn0sImxhc3ROYW1lIjp7InVpOmVtcHR5VmFsdWUiOiIiLCJ1aTphdXRvY29tcGxldGUiOiJnaXZlbi1uYW1lIn0sImFnZSI6eyJ1aTp3aWRnZXQiOiJ1cGRvd24iLCJ1aTp0aXRsZSI6IkFnZSBvZiBwZXJzb24iLCJ1aTpkZXNjcmlwdGlvbiI6IihlYXJ0aGlhbiB5ZWFyKSJ9LCJiaW8iOnsidWk6d2lkZ2V0IjoidGV4dGFyZWEifSwicGFzc3dvcmQiOnsidWk6d2lkZ2V0IjoicGFzc3dvcmQiLCJ1aTpoZWxwIjoiSGludDogTWFrZSBpdCBzdHJvbmchIn0sImRhdGUiOnsidWk6d2lkZ2V0IjoiYWx0LWRhdGV0aW1lIn0sInRlbGVwaG9uZSI6eyJ1aTpvcHRpb25zIjp7ImlucHV0VHlwZSI6InRlbCJ9fX0sInRoZW1lIjoiZGVmYXVsdCIsImxpdmVTZXR0aW5ncyI6eyJ2YWxpZGF0ZSI6dHJ1ZSwiZGlzYWJsZSI6ZmFsc2UsIm9taXRFeHRyYURhdGEiOmZhbHNlLCJsaXZlT21pdCI6ZmFsc2V9fQ==)
2. Open devtools inspector 
3. Search for `#root`

#### Expected behavior

There should only be one unique `id` (also useful for anchors)

#### Actual behavior

There are six `id=root` in the rendered markup

### Version

2.0.1

**Comment by SapienTech** [comment]
@jnachtigall  @epicfaace hi! getting this same issue. anything i can do to help move this along? this is a pretty big accessibility issue!

**Comment by jpsilva-daitan** [comment]
Same problem, any news about this?

**Comment by SapienTech** [comment]
@jpsilva-daitan I think I may have figured out the issue on my end, having to do with a missing value in the schema. I was missing `type: "object"` in the schema, can you try
```
...
spec: {
  ...
  schema: {
    type: 'object',
    ...
  }
}
```

**Comment by jpsilva-daitan** [comment]
> @jpsilva-daitan I think I may have figured out the issue on my end, having to do with a missing value in the schema. I was missing `type: "object"`, can you try
> 
> ```
> ...
> spec: {
>   ...
>   schema: {
>     type: 'object',
>     ...
>   }
> }
> ```

You are right, that was the problem.
Thanks =)

**Comment by n3ps** [comment]
@SapienTech @jpsilva-daitan I have `type: 'object'` but I don't see it having any effect to prevent the multiple `<fieldset id="root">`

**Comment by sepupl** [comment]
Found the same problem when array of objects, solutions was the same - to add "object" type to "items":

```
{
  "type": "object",
  "properties": {
    "someValues": {
      "type": "array",
      "items": {
        "type": "object",   // this is it
        "properties": {
          "prop1": {
            "type": "integer"
          }
        }
      }
    }
  }
}
```

**Comment by SleeplessByte** [comment]
Why is there a stalebot? Please remove this nonesense.

**Comment by heath-freenome** [comment]
@SleeplessByte we are a very small team of volunteers who have recently joined up to improve RJSF. Given that we don't have the history of the issues, it is possible they are either already fixed by someone or maybe aren't affecting our current users. As a result, we needed a way to reduce the noise in the issues list. Hence the stalebot, which is a common practice in open source software. We'd love help fixing this issue, especially if it is still impacting you. If you aren't able to help, at least allow us to make sense of what we've inherited.

**Comment by SleeplessByte** [comment]
@heath-freenome as a open source maintainer of over a dozen projects, I have all understanding for the _pain_, but triaging by closing everything using a stalebot isn't inviting anyone to put in more work. It means that whenever your team gets oversubscribed / too busy, new issues will always be closed as `wontfix`.

My recommendation would be:

- force a reproducible minimal example. I think this is perfectly fair especially because your time is as valuable as "ours"
- auto-close issues that do not link to a repository on github or gitlab (with a reproducible minimal example) or a link to the playground. I would recommend you add a message _why_ you're doing that. I personally like [this explanation](https://github.com/expo/expo/issues/18627#issuecomment-1307544077)
- if these are provided, you'd triage to check if the reproducibility is _actually_ minimal, and if so, add the label so it's never auto closed.

Issues don't go away because time has passed, and whilst it would be great, in general, issues don't get magically fixed without someone fixing them. I understand the reasoning, but in my 15 years of experience, the amount of issues that were resolved "accidentally" is single digit percentages compared to those that are tagged in PRs.

The problem right now is that you're banking on everyone who has ever opened an issue that: in the next x days (however you've configured stalebot) to go back in, reproduce the issue again, and compose a note. Do you really find that reasonable?

----------

As for this issue, pertaining to your question, the problem still occurs as reproduced on the playground:

![image](https://user-images.githubusercontent.com/1964376/218148074-d7ebc474-c017-4e03-b62c-bf76a79783c0.png)

**Comment by heath-freenome** [comment]
> The problem right now is that you're banking on everyone who has ever opened an issue that: in the next x days (however you've configured stalebot) to go back in, reproduce the issue again, and compose a note. Do you really find that reasonable?

Honestly, I've been having to bump issues in other repos that haven't been fixed and closed by stale bots in many of the repos I've filed issues in. A few others of us also have dealt with that. It's kind of where we got the idea to introduce a stalebot ourselves.

**Comment by heath-freenome** [comment]
> Issues don't go away because time has passed, and whilst it would be great, in general, issues don't get magically fixed without someone fixing them. I understand the reasoning, but in my 15 years of experience, the amount of issues that were resolved "accidentally" is single digit percentages compared to those that are tagged in PRs.

There are a category of issues that may have been fixed in this way because, when I undertook converting everything to Typescript I also tried to make sure that common code was refactored as best I could into the `utils` package or `core`. During that process I reduced the maintenance load by about 20%, and in the process may have fixed bugs in themes that were the result of inconsistent implementations. Anyway, I appreciate your thoughtful response

**Comment by heath-freenome** [comment]
> My recommendation would be:
> 
> * force a reproducible minimal example. I think this is perfectly fair especially because your time is as valuable as "ours"
> * auto-close issues that do not link to a repository on github or gitlab (with a reproducible minimal example) or a link to the playground. I would recommend you add a message _why_ you're doing that. I personally like [this explanation](https://github.com/expo/expo/issues/18627#issuecomment-1307544077)
> * if these are provided, you'd triage to check if the reproducibility is _actually_ minimal, and if so, add the label so it's never auto closed.

We'll take these recommendations under advisement! Again thanks for the thoughtful reply

**Comment by SleeplessByte** [comment]
> https://github.com/rjsf-team/react-jsonschema-form/issues/1824#issuecomment-1426435386

It's a solution that doesn't solve any issue or stop the bleeding. Given my involvement in many libraries I use myself (apart from the maintaining) results in a GitHub inbox of > 500 issues. From a purely selfish standpoint, it's horrible to experience the "90 day rotation" where all I do is confirm an issue still exist, but seems that you're taking the "tagged so never stale" approach, which is much better ;) I am just warning you / pointing out that this will severely decrease engagement down the line, but absolutely _understand_ and appreciate where you're coming from.

Feel free to personally tag me on other issues where you're unsure if it's still open where I'm in the thread. Happy to reconfirm once that those problems are still there.

**Comment by heath-freenome** [comment]
@jnachtigall So by simply adding `"type": "object"` to your schema, the issue goes away. I believe that this solution should be enough for you, no?
